stock_storage_type/models/stock_location.py

```diff
@@ -143,14 +143,20 @@ def _compute_leaf_location_ids(self):
 
     def _should_compute_will_contain_product_ids(self):
         return self.usage == "internal" and any(
-            location.do_not_mix_products
-            for location in self.allowed_location_storage_type_ids
+            location_storage_type.do_not_mix_products
+            for location_storage_type in self.allowed_location_storage_type_ids
         )
 
     def _should_compute_will_contain_lot_ids(self):
         return self.usage == "internal" and any(
-            location.do_not_mix_lots
-            for location in self.allowed_location_storage_type_ids
+            location_storage_type.do_not_mix_lots
+            for location_storage_type in self.allowed_location_storage_type_ids
+        )
+
+    def _should_compute_is_empty(self):
+        return self.usage == "internal" and any(
+            location_storage_type.only_empty
```

Comment:
@lmignon we worked on the same thing  at the same time :sob: 
#164 

However, I reached the conclusion that we do want to write on the field even if the value change when we pick or put in a location, to force transactions to put or pick sequentially with a retry. You can see my example in https://github.com/OCA/wms/pull/164/commits/9a3da83def369b9bc7f1975d50d3be2cb1ed9d6b

Reply:
Also, `stock_storage_type_buffer` needed to handle `location_is_empty` invalidation: https://github.com/OCA/wms/pull/164/commits/b0fcdf2879175e1c5205f57050a543ba7129399f

Reply:
oops, I was not referring to the correct line, what I meant is:
https://github.com/OCA/wms/pull/164/commits/402369071eceb08518c43fca5de9a94b6925e7bf#diff-29318822316101e435a8bb4690936de60502bf7b30abed00fac0ea9211f4d351R214-R215

Reply:
@guewen I propose to close this one since yours is more complete.